packages/ui/components/ui/tx/view/value/index.tsx

```diff
@@ -33,7 +33,7 @@ export const ValueViewComponent = ({
     const value = view.valueView.value;
     const metadata = view.valueView.value.metadata;
     const amount = value.amount ?? new Amount();
-    const exponent = getDisplayDenomExponent(metadata);
```

Comment:
Reviewers: I could use some help here.

When a user clicks the Undelegate button, the transaction approval dialog opens. We try to fetch the metadata for the undelegation tokens they'll be getting, but that will always come back empty, since the unbonding token type is now specific to the block height at which unbonding starts. That is, the unbonding token name is `unbonding_start_at_N_penumbravalid1...`, where `N` is the block height that unbonding started at.

There will _never_ be token metadata for an unbonding token until _after_ the transaction has been committed to the chain, since the token type is created when the transaction is committed.

But that means that the undelegation tokens are shown as "Unknown asset" in the transaction approval dialog:
<img width="512" alt="image" src="https://github.com/penumbra-zone/web/assets/1121544/9db34dbc-eef6-43c3-9fc3-401750ddef04">

Perhaps we should generate the unbonding token metadata on the fly here? If so, could someone point me in the right direction for how to do that? (That said, I'd love to get this PR merged before handling that, since _all_ delegation is blocked until this PR is merged.)

Reply:
Another wrinkle to all this: this issue, where the metadata request always returns an empty response for an undelegation token that doesn't exist yet, wasn't happening before when undelegation tokens were tied to epochs rather than block heights.

I don't know how that's possible. We should have had the same problem back then, right? That is:

1. We transition to a new epoch.
2. Someone goes to undelegate tokens from a validator.
3. Since no one else has undelegated tokens from that validator in this epoch yet, the metadata for `uunbonding_epoch_N_...` doesn't exist yet, so the metadata response is empty.
4. `getDisplayDenomExponent` throws, because it's being passed `undefined` metadata.

And yet, this wasn't happening before — no error was being thrown. It only started with this change, where unbonding tokens are tied to block height rather than epoch index.

I'm investigating this, but it's a bit concerning since I don't understand why it worked before!

Reply:
> Perhaps we should generate the unbonding token metadata on the fly here?

The `assetMetadataById` rpc method feels like a good place to generate metadata for this edge case so web apps have meaningful display data here.

> And yet, this wasn't happening before

Umm, no idea! 😅